server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1612,6 +1614,13 @@ public Ternary<Pair<List<? extends Host>, Integer>, List<? extends Host>, Map<Ho
             logger.debug("Hosts having capacity and suitable for migration: {}", suitableHosts);
         }
 
+        // Only list hosts of the same architecture as the source Host in a multi-arch zone
+        List<CPU.CPUArch> clusterArchs = ApiDBUtils.listZoneClustersArchs(vm.getDataCenterId());
+        if (CollectionUtils.isNotEmpty(clusterArchs) && clusterArchs.size() > 1) {
+            CPU.CPUArch hostArch = srcHost.getArch();
```

Comment:
Can we remove this line and compare directly on the line below without defining the variable?